wlauto/instrumentation/trace_cmd/__init__.py

```diff
@@ -237,35 +241,49 @@ def update_result(self, context):  # NOQA pylint: disable=R0912
-            # This is done host-side because the generated file is very large
-            if not os.path.isfile(local_trace_file):
-                self.logger.warning('Not generating trace.txt, as {} does not exist.'.format(OUTPUT_TRACE_FILE))
-            try:
+            # By default this is done host-side because the generated file is
+            # very large
+            if not self.report_on_target:
                 command = 'trace-cmd report {} > {}'.format(local_trace_file, local_txt_trace_file)
                 self.logger.debug(command)
-                process = subprocess.Popen(command, stderr=subprocess.PIPE, shell=True)
-                _, error = process.communicate()
-                if process.returncode:
-                    raise InstrumentError('trace-cmd returned non-zero exit code {}'.format(process.returncode))
-                if error:
-                    # logged at debug level, as trace-cmd always outputs some
-                    # errors that seem benign.
-                    self.logger.debug(error)
-                if os.path.isfile(local_txt_trace_file):
-                    context.add_iteration_artifact('txttrace', OUTPUT_TEXT_FILE, kind='export',
-                                                   description='trace-cmd generated ftrace dump.')
-                    self.logger.debug('Verifying traces.')
-                    with open(local_txt_trace_file) as fh:
-                        for line in fh:
-                            if 'EVENTS DROPPED' in line:
-                                self.logger.warning('Dropped events detected.')
-                                break
-                        else:
-                            self.logger.debug('Trace verified.')
+                if not os.path.isfile(local_trace_file):
+                    self.logger.warning('Not generating trace.txt, as {} does not exist.'.format(OUTPUT_TRACE_FILE))
                 else:
-                    self.logger.warning('Could not generate trace.txt.')
-            except OSError:
-                raise InstrumentError('Could not find trace-cmd. Please make sure it is installed and is in PATH.')
+                    try:
+                        process = subprocess.Popen(command, stderr=subprocess.PIPE, shell=True)
+                        _, error = process.communicate()
+                        if process.returncode:
+                            raise InstrumentError('trace-cmd returned non-zero exit code {}'.format(process.returncode))
+                        if error:
+                            # logged at debug level, as trace-cmd always outputs some
+                            # errors that seem benign.
+                            self.logger.debug(error)
+                    except OSError:
+                        raise InstrumentError('Could not find trace-cmd. Please make sure it is installed and is in PATH.')
+
+            else:
```

Comment:
Actually thinking about it, that whole construct is kinda large and unwieldly; might be worth factoring parts of it out into helper methods; something like


    if self.report_on_target:
        self._generate_report_on_target()
    else: 
        self._generate_report_on_host()